src/smt/smt_engine.cpp

```diff
@@ -1419,9 +1419,20 @@ UnsatCore SmtEngine::getUnsatCoreInternal()
   // generate with new proofs
   PropEngine* pe = getPropEngine();
   Assert(pe != nullptr);
-  Assert(pe->getProof() != nullptr);
-  std::shared_ptr<ProofNode> pfn = d_pfManager->getFinalProof(
-      pe->getProof(), *d_asserts, *d_definedFunctions);
+
+  std::shared_ptr<ProofNode> pepf;
+  if (options::unsatCores()
+      && options::unsatCoresMode() == options::UnsatCoresMode::ASSUMPTIONS)
```

Comment:
```suggestion
  if (options::unsatCoresMode() == options::UnsatCoresMode::ASSUMPTIONS)
```

Reply:
After this change I'd actually inline the ite, but as you wish.

Reply:
Doesn't really make it more readable, I'll keep the if/else.